src/python/oftest/testutils.py

```diff
@@ -353,7 +353,16 @@ def simple_icmpv6_packet(pktlen=100,
                          ipv6_hlim=64,
                          ipv6_fl=0,
                          icmp_type=8,
-                         icmp_code=0):
+                         icmp_code=0,
+                         icmp_data='',
+                         prefix_opt = False,
+                         source_ll = False,
+                         ll_addr = '66:6f:df:2d:7c:9c',
+                         ipv6_prefix='fd00:141:64:1::',
+                         R_bit = 0,
+                         S_bit = 0,
+                         target = '::'
+                         ):
```

Comment:
instead of adding lots of additional parameters to this function, how about adding utility functions to construct the body of the icmpv6 message and just adding an "icmp_body" or "icmp_data" parameter?  you can then call simple_icmpv6_packet with the body and use the '/' operator to add the body to pkt.
note also that the additional parameters need to be documented in the python docstring below.